app/queries/servers/channel.ts

```diff
@@ -536,6 +536,19 @@ export function observeMyChannelMentionCount(database: Database, teamId?: string
         );
 }
 
+export function observeMyChannelUnreads(database: Database, teamId: string) {
+    const myChannels = queryMyChannelsByTeam(database, teamId).observeWithColumns(['is_unread']);
+    const notifyProps = observeAllMyChannelNotifyProps(database);
+    return myChannels.pipe(
+        combineLatestWith(notifyProps),
+        switchMap(([mycs, notify]) => of$(mycs.reduce((acc, v) => {
+            const isMuted = notify?.[v.id]?.mark_unread === 'mention';
+            return acc || (v.isUnread && !isMuted);
+        }, false))),
```

Comment:
I feel some comments to explain the logic could be beneficial for people that don't usually deal with mobile code, and all the rxjs syntax involved.

Reply:
Maybe I am falling into a bit of knowledge curse. What is hard to understand here that has to deal with rxjs?

From the rxjs part, we only have the `switchMap` and the `of$`. Both are a bit alien for anyone that have never used rxjs, but they are all over the code, so it doesn't make sense to document them (I think).

The rest of the code, is a simple list reduce, that checks the business logic of mattermost around when a message is considered unread (is actually unread, and is not muted).

So... not sure if I am missing something.